apps/explorer/src/utils/getAmout.ts

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+    getPaySuiTransaction,
+    getPayTransaction,
+    getTransferSuiTransaction,
+    getTransferObjectTransaction,
+    getTransactionKindName,
+} from '@mysten/sui.js';
+
+import type {
+    SuiTransactionKind,
+    TransactionEffects,
+    SuiEvent,
+} from '@mysten/sui.js';
+
+function notEmpty<TValue>(value: TValue | null | undefined): value is TValue {
+    if (value === null || value === undefined) return false;
+    return true;
+}
+
+const getCoinType = (
+    txEffects: TransactionEffects,
+    address: string
+): string | null => {
+    const events = txEffects?.events || [];
+    const coinType = events
+        ?.map((event: SuiEvent) => {
+            const data = Object.values(event).find(
+                (itm) => itm?.owner?.AddressOwner === address
+            );
+            return data?.coinType;
+        })
+        .filter(notEmpty);
```

Comment:
You can just `.filter(Boolean)` although you'd need to refine the types because TypeScript doesn't realize what that does.